### ISSUE TITLE
Include dependency verification in scorecard

### ISSUE THREAD

**Issue by melix** [ossf/scorecard#664]
[Dependency verification](https://docs.gradle.org/current/userguide/dependency_verification.html), as implemented by Gradle for example, allows verifying both checksums and signatures of dependencies _actually used in a build_. It is, IMHO, significantly more important than using an automated dependency upgrade tool, in comparison.

It would be great if this was actually considered in the score, since we strongly encourage users to enable dependency verification as a tool to reduce the risks of supply chain attacks.

**Comment by laurentsimon** [comment]
Thanks for the suggestion. This is definitely something we should consider supporting long-term.

We've been hesitant to add any check around signature verification so far, mostly because there does not seem to be a well rounded story around key management/discoverability/revokation. We need to take a second look.

Related link for npm: https://docs.npmjs.com/verifying-the-pgp-signature-for-a-package-from-the-npm-public-registry

There are other ongoing efforts like [sigstore](https://sigstore.dev/) that are relevant.

Any feedback, suggestion or ideas are welcome!

**Comment by naveensrinivasan** [comment]
I think we can work on Dependency verification check 
https://docs.gradle.org/current/userguide/dependency_verification.html, probably  the signature verification later.


```
<?xml version="1.0" encoding="UTF-8"?>
<verification-metadata>
   <configuration>
      <verify-metadata>true</verify-metadata>
      <verify-signatures>false</verify-signatures>
    </configuration>
</verification-metadata>
```

@laurentsimon Thoughts?

**Comment by laurentsimon** [comment]
> I think we can work on Dependency verification check https://docs.gradle.org/current/userguide/dependency_verification.html, probably the signature verification later.
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <verification-metadata>
>    <configuration>
>       <verify-metadata>true</verify-metadata>
>       <verify-signatures>false</verify-signatures>
>     </configuration>
> </verification-metadata>
> ```
> 
> @laurentsimon Thoughts?

Agreed it's a great idea. We can start with the integrity part and implement it in the Pinned Dependency check. Is it what you had in mind too, @naveensrinivasan ?

**Comment by naveensrinivasan** [comment]
> > I think we can work on Dependency verification check https://docs.gradle.org/current/userguide/dependency_verification.html, probably the signature verification later.
> > ```
> > <?xml version="1.0" encoding="UTF-8"?>
> > <verification-metadata>
> >    <configuration>
> >       <verify-metadata>true</verify-metadata>
> >       <verify-signatures>false</verify-signatures>
> >     </configuration>
> > </verification-metadata>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @laurentsimon Thoughts?
> 
> Agreed it's a great idea. We can start with the integrity part and implement it in the Pinned Dependency check. Is it what you had in mind too, @naveensrinivasan ?

Yes, that's what I had in mind. For the first pass we check if the file exists and if it has `<verify-metadata>true</verify-metadata>` flag enabled.

**Comment by naveensrinivasan** [comment]
> > > I think we can work on Dependency verification check https://docs.gradle.org/current/userguide/dependency_verification.html, probably the signature verification later.
> > > ```
> > > <?xml version="1.0" encoding="UTF-8"?>
> > > <verification-metadata>
> > >    <configuration>
> > >       <verify-metadata>true</verify-metadata>
> > >       <verify-signatures>false</verify-signatures>
> > >     </configuration>
> > > </verification-metadata>
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > @laurentsimon Thoughts?
> > 
> > 
> > Agreed it's a great idea. We can start with the integrity part and implement it in the Pinned Dependency check. Is it what you had in mind too, @naveensrinivasan ?
> 
> Yes, that's what I had in mind. For the first pass we check if the file exists and if it has `<verify-metadata>true</verify-metadata>` flag enabled.

@laurentsimon  What about the score calculation? How should this affect the score?

**Comment by justaugustus** [comment]
Discussed in 5/16 meeting:
No gradle experts on the call, would require research, potentially out-of-scope for the project (but nice to have)
